app/Http/Controllers/PeopleController.php

```diff
@@ -281,116 +257,77 @@ public function delete(Request $request, $contactId)
 
         $contact->delete();
 
-        $request->session()->flash('success', trans('people.people_delete_success'));
-
-        return redirect()->route('people.index');
+        return redirect()->route('people.index')
+            ->with('success', trans('people.people_delete_success'));
     }
 
     /**
      * Show the Edit work view.
+     *
      * @param  Request $request
-     * @return View
+     * @param Contact $contact
+     * @return \Illuminate\Http\Response
      */
-    public function editWork(Request $request, $contactId)
+    public function editWork(Request $request, Contact $contact)
     {
-        $contact = Contact::findOrFail($contactId);
-
-        if ($contact->account_id !== Auth::user()->account_id) {
-            return redirect()->route('people.index');
-        }
-
-        $data = [
-            'contact' => $contact,
-        ];
-
-        return view('people.dashboard.work.edit', $data);
+        return view('people.dashboard.work.edit')
+            ->withContact($contact);
     }
 
     /**
      * Save the work information
-     * @param int $id
-     * @return
+     *
+     * @param Request $request
+     * @param Contact $contact
+     * @return \Illuminate\Http\Response
      */
-    public function updateWork(Request $request, $contactId)
+    public function updateWork(Request $request, Contact $contact)
     {
-        $contact = Contact::findOrFail($contactId);
-
-        if ($contact->account_id !== Auth::user()->account_id) {
-            return redirect()->route('people.index');
-        }
-
         $job = $request->input('job');
         $company = $request->input('company');
+        $linkedin = $request->input('linkedin');
 
-        if ($job != '') {
-            $contact->job = $job;
-        } else {
-            $contact->job = null;
-        }
+        $contact->job = ! empty($job) ? $job : null;
 
-        if ($company != '') {
-            $contact->company = $company;
-        } else {
-            $contact->company = null;
-        }
+        // There is no company field in the database.
```

Comment:
There actually is. Check https://github.com/monicahq/monica/blob/master/database/migrations/2017_06_13_195740_add_company_to_contacts.php.